example/wallet.ts

```diff
@@ -31,179 +31,356 @@ import {
 
-    this.accounts = [new MyWalletAccount(this.signer)]
+    // Update our Wallet object to know that we are connected to this new signer. 
+    this.accounts = [new MyWalletAccount(this.signer)];
   }
 
+  /**
+   * REVISION - Implement this function using your Wallet. 
+   * 
+   * Look up the account info for the currently connected wallet address on the chosen network. 
+   * 
+   * @returns Return account info.
+   */
   account: AptosGetAccountMethod = async (): Promise<AccountInfo> => {
     const account = new AccountInfo({
       address: this.signer.accountAddress,
-      publicKey: this.signer.publicKey
-    })
-    return Promise.resolve(account)
-  }
+      publicKey: this.signer.publicKey,
+    });
+    return Promise.resolve(account);
+  };
 
-  connect: AptosConnectMethod = async (): Promise<UserResponse<AccountInfo>> => {
+  /**
+   * REVISION - Implement this function using your Wallet. 
+   * 
+   * Connect an account using this Wallet. 
+   * This must wait for the user to sign in to the Wallet provider and confirm they are ok sharing
+   * details with the dapp.
+   * 
+   * For demonstration purposes, this template example assumes the user is using the account generated in `signer`
+   * and assumes the user approved letting the dapp use the account information.
+   * 
+   * Your implmentation should include a way to track which account was just connected. This likely will involve
+   * setting the `this.accounts` variable. 
+   * 
+   * @returns Whether the user approved connecting their account, and account info.
+   * @throws Error when unable to connect to the Wallet provider.
+   */
+  connect: AptosConnectMethod = async (): Promise<
+    UserResponse<AccountInfo>
+  > => {
+    // THIS LOGIC SHOULD BE REPLACED. IT IS FOR EXAMPLE PURPOSES ONLY.    
     try {
       await this.aptos.fundAccount({
         accountAddress: this.signer.accountAddress,
-        amount: 1_000_000
-      })
+        amount: 1_000_000,
+      });
       const account = new AccountInfo({
         address: this.signer.accountAddress,
-        publicKey: this.signer.publicKey
-      })
+        publicKey: this.signer.publicKey,
+      });
       return {
         status: UserResponseStatus.APPROVED,
-        args: account
-      }
+        args: account,
+      };
     } catch (e) {
       return {
         status: UserResponseStatus.REJECTED
-      }
+      }    
     }
-  }
+  };
 
+  /**
+   * REVISION - Implement this function using your Wallet. 
+   * 
+   * Return the name, chainId, and url of the network connection your wallet is using to connect to the Aptos chain.
+   * 
+   * @returns Which network the connected Wallet is pointing to.
+   */
   network: AptosGetNetworkMethod = async (): Promise<NetworkInfo> => {
-    const network = await this.aptos.getLedgerInfo()
+    // THIS LOGIC SHOULD BE REPLACED. IT IS FOR EXAMPLE PURPOSES ONLY.    
+    // You may use getLedgerInfo() to determine which ledger your Wallet is connected to.
```

Comment:
nit

for mainnet/testnet it is encouraged to not make a server call as the chain_id is fixed

Reply:
Updated with a "REVISION" note :)